src/provider/provider_tracking.c

```diff
@@ -330,6 +331,8 @@ static umf_result_t trackingAllocationMerge(void *hProvider, void *lowPtr,
 
 err:
     utils_mutex_unlock(&provider->hTracker->splitMergeMutex);
+    assert(0);
```

Comment:
Do we need this assert? I think it is OK if Memory provider does not support Merge  operation.